webapp/src/main/java/com/box/l10n/mojito/service/ai/openai/OpenAILLMService.java

```diff
@@ -93,6 +117,54 @@ public AICheckResponse executeAIChecks(AICheckRequest aiCheckRequest) {
     return aiCheckResponse;
   }
 
+  @Override
+  @Timed("OpenAILLMService.translate")
+  public String translate(
+      TMTextUnit tmTextUnit, String sourceBcp47Tag, String targetBcp47Tag, AIPrompt prompt) {
+    logger.debug(
+        "Translating text unit {} from {} to {} using prompt {}",
+        tmTextUnit.getId(),
+        sourceBcp47Tag,
+        targetBcp47Tag,
+        prompt.getId());
+    String systemPrompt =
+        getTranslationFormattedPrompt(
+            prompt.getSystemPrompt(), tmTextUnit, sourceBcp47Tag, targetBcp47Tag);
+    String userPrompt =
+        getTranslationFormattedPrompt(
+            prompt.getUserPrompt(), tmTextUnit, sourceBcp47Tag, targetBcp47Tag);
+
+    OpenAIClient.ChatCompletionsRequest chatCompletionsRequest =
+        buildChatCompletionsRequest(
+            prompt, systemPrompt, userPrompt, prompt.getContextMessages(), false);
+
+    return Mono.fromCallable(
+            () -> {
+              OpenAIClient.ChatCompletionsResponse chatCompletionsResponse =
+                  openAIClient.getChatCompletions(chatCompletionsRequest).join();
+              String response = chatCompletionsResponse.choices().getFirst().message().content();
```

Comment:
Given we only support single-choice (n=1) initially - should we assert or log an error on that (having a single choice in the response) to prevent a misconfigured client asking for n=5 and spending more when we would only use the first response anyways? Thoughts?

Reply:
My concern would be that if we add support for n>1 we might forget to update this and incur extra cost without noticing, as we're not likely to notice the warnings.

Any thoughts on having a hard fail in this scenario here instead of a warning?

Reply:
I can add a hard failure instead